rir/src/utils/configurations.cpp

```diff
@@ -109,10 +109,11 @@ void Configurations::defaultOptimizations() {
     // ==== Phase 4) Final round of default opts
     for (size_t i = 0; i < 3; ++i) {
         addDefaultOpt();
+
         optimizations.push_back(new pir::CleanupCheckpoints());
-    }
+        optimizations.push_back(new pir::LoopInvariant());
```

Comment:
Yes, I put it late because we were discussing that the loop detection may be slow. I add it now near hoistInstruction with which this opt should be integrated.